pyomo/contrib/community_detection/tests/test_detection.py

```diff
@@ -30,8 +30,8 @@
 from pyomo.solvers.tests.models.LP_inactive_index import LP_inactive_index
 from pyomo.solvers.tests.models.SOS1_simple import SOS1_simple
 
-if matplotlib_available:
-    matplotlib.use('Agg')  # added to avoid $DISPLAY errors on Travis (from parmest)
+matplotlib.use('Agg')  # added to avoid $DISPLAY errors on Travis (from parmest)
```

Comment:
Ironically, this still needs to be guarded by:
```python
if matplotlib_available:
    matplotlib.use('Agg')  # added to avoid $DISPLAY errors on Travis (from parmest)
```

Because we are in the test harness, it is OK to trigger the import check for optional modules.  Further, we need to guard the call to `.use`, as when that is parsed, the `DeferredImportModule` (i.e., `matplotlib`) will trigger the import and then if `matplotlib` is not available, the attempt to return the `.use` attribute will generate an exception.

Reply:
Okay I see - making the changes now!